org.eclipse.xtext.xbase/deprecated/org/eclipse/xtext/xbase/formatting/HiddenLeafAccess.java

```diff
@@ -0,0 +1,169 @@
+					} else {
+						if (!Character.isWhitespace(c)) {
+							comment = true;
+						}
+					}
+				}
+			}
+			if (comment) {
+				if (!(IterableExtensions.last(result.getLeafs()) instanceof WhitespaceInfo)) {
+					result.getLeafs().add(new WhitespaceInfo(result, null, 0, node.getOffset()));
+				}
+				result.getLeafs().add(new CommentInfo(result, node, newLines, trailing));
+			} else {
+				result.getLeafs().add(new WhitespaceInfo(result, node, newLines, node.getOffset()));
+			}
+			if (newLines > 0) {
+				trailing = false;
+			}
+		}
+		if (!(IterableExtensions.last(result.getLeafs()) instanceof WhitespaceInfo)) {
+			int whitespaceOffset = 0;
+			if (result.getLeafs().isEmpty()) {
+				whitespaceOffset = offset;
+			} else {
+				whitespaceOffset = IterableExtensions.last(result.getLeafs()).getNode().getEndOffset();
+			}
+			result.getLeafs().add(new WhitespaceInfo(result, null, 0, whitespaceOffset));
+		}
+		return result;
+	}
+
+	public HiddenLeafs getHiddenLeafsAfter(INode node) {
+		ILeafNode start = findPreviousLeaf(node, l -> !l.isHidden());
+		if (start != null) {
+			return newHiddenLeafs(start.getEndOffset(), findNextHiddenLeafs(start));
+		} else {
+			offset = 0;
+			if (node != null) {
+				offset = node.getOffset();
+			}
+			return new HiddenLeafs(offset);
+		}
+	}
+
+	protected List<ILeafNode> findNextHiddenLeafs(INode node) {
+		List<ILeafNode> result = new ArrayList<>();
+		NodeIterator ni = new NodeIterator(node);
+		while (ni.hasNext()) {
+			INode next = ni.next();
+			if (next instanceof ILeafNode) {
+				if (((ILeafNode) next).isHidden()) {
+					result.add((ILeafNode) next);
+				} else {
+					return result;
+				}
+			}
+		}
+		return result;
+	}
+
+	protected ILeafNode findPreviousLeaf(INode node, Function1<? super ILeafNode, ? extends Boolean> matches) {
+		INode current = node;
+		while (current instanceof ICompositeNode) {
+			current = ((ICompositeNode) current).getLastChild();
+		}
+		if (current instanceof ILeafNode && matches.apply((ILeafNode) current).booleanValue()) {
+			return (ILeafNode) current;
+		}
+		if (current != null) {
+			NodeIterator ni = new NodeIterator(current);
+			while (ni.hasPrevious()) {
+				INode previous = ni.previous();
+				if (previous instanceof ILeafNode && matches.apply((ILeafNode) previous).booleanValue()) {
+					return (ILeafNode) previous;
+				}
+			}
+		}
+		return null;
+	}
+
+	protected List<ILeafNode> findPreviousHiddenLeafs(INode node) {
+		INode current = node;
+		while (current instanceof ICompositeNode) {
+			current = ((ICompositeNode) current).getLastChild();
+		}
+		List<ILeafNode> result = new ArrayList<>();
+		if (current != null) {
+			NodeIterator ni = new NodeIterator(current);
+			while (ni.hasPrevious()) {
+				INode previous = ni.previous();
+				if (!Objects.equal(previous, current) && previous instanceof ILeafNode) {
+					if (((ILeafNode) previous).isHidden()) {
+						result.add((ILeafNode) previous);
+					} else { // if(!result.empty)
```

Comment:
Seems like a pretty useless comment to me

Reply:
did not want to loose it